lib/rspec-steps/dsl.rb

```diff
@@ -0,0 +1,30 @@
+require 'rspec-steps/builder'
+require 'rspec-steps/describer'
+
+module RSpec::Steps
+  def self.warnings
+    @warnings ||= Hash.new do |h,warning|
+      puts warning #should be warn, but RSpec complains
+      h[warning] = true
+    end
+  end
+
+  SharedSteps = {}
+
+  module DSL
+    def steps(*args, &block)
+      describer = Describer.new(args, {:caller => caller}, &block)
+      builder = Builder.new(describer)
+
+      builder.build_example_group
+    end
+
+    def shared_steps(*args, &block)
+      name = args.first
```

Comment:
There seems to be an identical mehtod on the describer for shared_steps -- can you explain why it has to be in both places?